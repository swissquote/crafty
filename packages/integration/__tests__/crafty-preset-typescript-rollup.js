/* global describe, it, expect, jest */

const testUtils = require("../utils");

// Add a high timeout because of https://github.com/facebook/jest/issues/8942
// Tests would be unreliable if they timeout >_<
jest.setTimeout(30000);

// node-forge 0.6.33 doesn't work with jest.
// but selfsigned is fixed on this version
// as we don't use it for now, we can simply mock it
// https://github.com/jfromaniello/selfsigned/issues/16
jest.mock("node-forge");

it("Works with rollup", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript-rollup/compiles"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js.map")).toBeTruthy();
  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myTSBundle.min.js")
  ).toMatchSnapshot();
});

it("Deletes rollup terser plugin using crafty.config.js", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript-rollup/compiles-no-terser"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js.map")).toBeTruthy();
  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myTSBundle.min.js")
  ).toMatchSnapshot();
});

it("Keeps imports unresolved for Babel Runtime", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript-rollup/compiles-import-runtime"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js.map")).toBeTruthy();
  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myTSBundle.min.js")
  ).toMatchSnapshot();
});

it("Fails gracefully on broken markup", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript-rollup/fails"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints TypeScript with rollup", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript-rollup/lints"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});
