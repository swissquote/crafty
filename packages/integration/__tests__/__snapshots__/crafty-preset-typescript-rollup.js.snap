// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deletes rollup terser plugin using crafty.config.js 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/compiles-no-terser/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
[__:__:__] Finished 'js_myTSBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Deletes rollup terser plugin using crafty.config.js 2`] = `
"function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError(\\"Cannot call a class as a function\\");
  }
}

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if (\\"value\\" in descriptor) descriptor.writable = true;
    Object.defineProperty(target, descriptor.key, descriptor);
  }
}

function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  return Constructor;
}

var C = /*#__PURE__*/function () {
  function C() {
    _classCallCheck(this, C);

    /* eslint-disable-next-line no-console */
    console.log(\\"Something happens here\\");
  }

  _createClass(C, [{
    key: \\"someMethod\\",
    value: function someMethod() {
      /* eslint-disable-next-line no-alert */
      alert(\\"yes, alert still exists\\");
    }
  }]);

  return C;
}();

var D = /*#__PURE__*/function () {
  function D() {
    _classCallCheck(this, D);

    /* eslint-disable-next-line no-console */
    console.log(\\"Something else happens here\\");
  }

  _createClass(D, [{
    key: \\"someMethod\\",
    value: function someMethod() {
      /* eslint-disable-next-line no-alert */
      alert(\\"yes, alert still exists\\");
    }
  }]);

  return D;
}();

export { C, D };
//# sourceMappingURL=myTSBundle.min.js.map
"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly

__PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/fails/js/typescript.ts
  14:0  error  Parsing error: Unexpected token. A constructor, method, accessor, or property was expected

✖ 1 problem (1 error, 0 warnings)

[js_myTSBundle] (!) Errors were found
[__:__:__] 'js_myTSBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;

exports[`Keeps imports unresolved for Babel Runtime 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/compiles-import-runtime/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
[__:__:__] Finished 'js_myTSBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Keeps imports unresolved for Babel Runtime 2`] = `
"import _classCallCheck from '@babel/runtime/helpers/classCallCheck';
import _createClass from '@babel/runtime/helpers/createClass';

var C = /*#__PURE__*/function () {
  function C() {
    _classCallCheck(this, C);

    /* eslint-disable-next-line no-console */
    console.log(\\"Something happens here\\");
  }

  _createClass(C, [{
    key: \\"someMethod\\",
    value: function someMethod() {
      /* eslint-disable-next-line no-alert */
      alert(\\"yes, alert still exists\\");
    }
  }]);

  return C;
}();

var D = /*#__PURE__*/function () {
  function D() {
    _classCallCheck(this, D);

    /* eslint-disable-next-line no-console */
    console.log(\\"Something else happens here\\");
  }

  _createClass(D, [{
    key: \\"someMethod\\",
    value: function someMethod() {
      /* eslint-disable-next-line no-alert */
      alert(\\"yes, alert still exists\\");
    }
  }]);

  return D;
}();

export { C, D };
//# sourceMappingURL=myTSBundle.min.js.map
"
`;

exports[`Lints TypeScript with rollup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/lints/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly

__PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/lints/js/Component.ts
  1:1  error  Replace \`⏎export·function·test(a:\` with \`export·function·test(a:·\`  @swissquote/swissquote/prettier/prettier
  3:1  error  Replace \`···return·a·+···\` with \`··return·a·+\`                      @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/lints/js/SomeLibrary.ts
   1:1  error  Delete \`⏎\`                                        @swissquote/swissquote/prettier/prettier
   3:3  error  Delete \`··\`                                       @swissquote/swissquote/prettier/prettier
   4:1  error  Replace \`········\` with \`····\`                    @swissquote/swissquote/prettier/prettier
   5:1  error  Replace \`····\` with \`··\`                          @swissquote/swissquote/prettier/prettier
   7:1  error  Delete \`··\`                                       @swissquote/swissquote/prettier/prettier
   8:3  error  Insert \`··\`                                       @swissquote/swissquote/prettier/prettier
   9:1  error  Delete \`··\`                                       @swissquote/swissquote/prettier/prettier
  11:3  error  Replace \`··divide(a,·b)⏎···\` with \`divide(a,·b)\`  @swissquote/swissquote/prettier/prettier
  13:1  error  Replace \`········\` with \`····\`                    @swissquote/swissquote/prettier/prettier
  14:1  error  Delete \`····\`                                     @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/lints/js/script.ts
   1:1   error    Replace \`⏎import·{\` with \`import·{·\`         @swissquote/swissquote/prettier/prettier
   6:3   error    Replace \`··constructor·\` with \`constructor\`  @swissquote/swissquote/prettier/prettier
   7:5   warning  Unexpected console statement                 no-console
   8:2   error    Insert \`·\`                                   @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                 no-console
  11:17  error    Replace \`·test(·\` with \`test(\`               @swissquote/swissquote/prettier/prettier
  13:16  error    Replace \`·(·)·=>⏎···\` with \`()·=>\`           @swissquote/swissquote/prettier/prettier
  17:7   warning  Unexpected console statement                 no-console
  18:6   error    Replace \`·,·300)\` with \`,·300);\`             @swissquote/swissquote/prettier/prettier

✖ 21 problems (18 errors, 3 warnings)
  18 errors and 0 warnings potentially fixable with the \`--fix\` option.

[js_myBundle] (!) Errors were found
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;

exports[`Works with rollup 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
[__:__:__] Finished 'js_myTSBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Works with rollup 2`] = `
"function e(e,n){if(!(e instanceof n))throw new TypeError(\\"Cannot call a class as a function\\")}function n(e,n){for(var t=0;t<n.length;t++){var o=n[t];o.enumerable=o.enumerable||!1,o.configurable=!0,\\"value\\"in o&&(o.writable=!0),Object.defineProperty(e,o.key,o)}}function t(e,t,o){return t&&n(e.prototype,t),o&&n(e,o),e}var o=/*#__PURE__*/function(){function n(){e(this,n),console.log(\\"Something happens here\\")}return t(n,[{key:\\"someMethod\\",value:function(){alert(\\"yes, alert still exists\\")}}]),n}();var r=/*#__PURE__*/function(){function n(){e(this,n),console.log(\\"Something else happens here\\")}return t(n,[{key:\\"someMethod\\",value:function(){alert(\\"yes, alert still exists\\")}}]),n}();export{o as C,r as D};
//# sourceMappingURL=myTSBundle.min.js.map
"
`;
